In [1]:
# Generic inputs for most ML tasks
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
# This is new
from sklearn.linear_model import LogisticRegression
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
from sklearn.ensemble import RandomForestRegressor

pd.options.display.float_format = '{:,.2f}'.format

# setup interactive notebook mode
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

from IPython.display import display, HTML

In [2]:
# fetch data and print the first five elements and column name

airline_data = pd.read_csv('Invistico_Airline.csv')
airline_data.head()

satisfaction  Gender   Customer Type  Age   Type of Travel     Class  \
0    satisfied  Female  Loyal Customer   65  Personal Travel       Eco   
1    satisfied    Male  Loyal Customer   47  Personal Travel  Business   
2    satisfied  Female  Loyal Customer   15  Personal Travel       Eco   
3    satisfied  Female  Loyal Customer   60  Personal Travel       Eco   
4    satisfied  Female  Loyal Customer   70  Personal Travel       Eco   

   Flight Distance  Seat comfort  Departure/Arrival time convenient  \
0              265             0                                  0   
1             2464             0                                  0   
2             2138             0                                  0   
3              623             0                                  0   
4              354             0                                  0   

   Food and drink  ...  Online support  Ease of Online booking  \
0               0  ...               2                       3   
1               0  ...               2                       3   
2               0  ...               2                       2   
3               0  ...               3                       1   
4               0  ...               4                       2   

   On-board service  Leg room service  Baggage handling  Checkin service  \
0                 3                 0                 3                5   
1                 4                 4                 4                2   
2                 3                 3                 4                4   
3                 1                 0                 1                4   
4                 2                 0                 2                4   

   Cleanliness  Online boarding  Departure Delay in Minutes  \
0            3                2                           0   
1            3                2                         310   
2            4                2                           0   
3            1                3                           0   
4            2                5                           0   

   Arrival Delay in Minutes  
0                      0.00  
1                    305.00  
2                      0.00  
3                      0.00  
4                      0.00  

[5 rows x 23 columns]

In [3]:
column_names = airline_data.columns
print(column_names)

Index(['satisfaction', 'Gender', 'Customer Type', 'Age', 'Type of Travel',
       'Class', 'Flight Distance', 'Seat comfort',
       'Departure/Arrival time convenient', 'Food and drink', 'Gate location',
       'Inflight wifi service', 'Inflight entertainment', 'Online support',
       'Ease of Online booking', 'On-board service', 'Leg room service',
       'Baggage handling', 'Checkin service', 'Cleanliness', 'Online boarding',
       'Departure Delay in Minutes', 'Arrival Delay in Minutes'],
      dtype='object')


In [4]:
airline_data.dtypes

satisfaction                          object
Gender                                object
Customer Type                         object
Age                                    int64
Type of Travel                        object
Class                                 object
Flight Distance                        int64
Seat comfort                           int64
Departure/Arrival time convenient      int64
Food and drink                         int64
Gate location                          int64
Inflight wifi service                  int64
Inflight entertainment                 int64
Online support                         int64
Ease of Online booking                 int64
On-board service                       int64
Leg room service                       int64
Baggage handling                       int64
Checkin service                        int64
Cleanliness                            int64
Online boarding                        int64
Departure Delay in Minutes             int64
Arrival De

In [5]:
#Check NaN values

airline_data.isna()
airline_data.isna().sum()

satisfaction  Gender  Customer Type    Age  Type of Travel  Class  \
0              False   False          False  False           False  False   
1              False   False          False  False           False  False   
2              False   False          False  False           False  False   
3              False   False          False  False           False  False   
4              False   False          False  False           False  False   
...              ...     ...            ...    ...             ...    ...   
129875         False   False          False  False           False  False   
129876         False   False          False  False           False  False   
129877         False   False          False  False           False  False   
129878         False   False          False  False           False  False   
129879         False   False          False  False           False  False   

        Flight Distance  Seat comfort  Departure/Arrival time convenient  \
0                 False         False                              False   
1                 False         False                              False   
2                 False         False                              False   
3                 False         False                              False   
4                 False         False                              False   
...                 ...           ...                                ...   
129875            False         False                              False   
129876            False         False                              False   
129877            False         False                              False   
129878            False         False                              False   
129879            False         False                              False   

        Food and drink  ...  Online support  Ease of Online booking  \
0                False  ...           False                   False   
1                False  ...           False                   False   
2                False  ...           False                   False   
3                False  ...           False                   False   
4                False  ...           False                   False   
...                ...  ...             ...                     ...   
129875           False  ...           False                   False   
129876           False  ...           False                   False   
129877           False  ...           False                   False   
129878           False  ...           False                   False   
129879           False  ...           False                   False   

        On-board service  Leg room service  Baggage handling  Checkin service  \
0                  False             False             False            False   
1                  False             False             False            False   
2                  False             False             False            False   
3                  False             False             False            False   
4                  False             False             False            False   
...                  ...               ...               ...              ...   
129875             False             False             False            False   
129876             False             False             False            False   
129877             False             False             False            False   
129878             False             False             False            False   
129879             False             False             False            False   

        Cleanliness  Online boarding  Departure Delay in Minutes  \
0             False            False                       False   
1             False            False                       False   
2             False            False                       False   
3             False            False                       False   
4   

satisfaction                           0
Gender                                 0
Customer Type                          0
Age                                    0
Type of Travel                         0
Class                                  0
Flight Distance                        0
Seat comfort                           0
Departure/Arrival time convenient      0
Food and drink                         0
Gate location                          0
Inflight wifi service                  0
Inflight entertainment                 0
Online support                         0
Ease of Online booking                 0
On-board service                       0
Leg room service                       0
Baggage handling                       0
Checkin service                        0
Cleanliness                            0
Online boarding                        0
Departure Delay in Minutes             0
Arrival Delay in Minutes             393
dtype: int64

In [6]:
#drop NaN values
airline_data.dropna(subset=['Arrival Delay in Minutes'],inplace=True)

In [7]:
#checking Nan Values after dropping them
airline_data.isna().sum()

satisfaction                         0
Gender                               0
Customer Type                        0
Age                                  0
Type of Travel                       0
Class                                0
Flight Distance                      0
Seat comfort                         0
Departure/Arrival time convenient    0
Food and drink                       0
Gate location                        0
Inflight wifi service                0
Inflight entertainment               0
Online support                       0
Ease of Online booking               0
On-board service                     0
Leg room service                     0
Baggage handling                     0
Checkin service                      0
Cleanliness                          0
Online boarding                      0
Departure Delay in Minutes           0
Arrival Delay in Minutes             0
dtype: int64

#### Answer - The **"Arrival Delay in Minutes"** column has **393** NaN values in the dataset which is dropped

In [8]:
#Counting the number of rows in the dataset after removing the NaN rows
airline_data.shape

(129487, 23)

#### Answer - After dropping NaN values, the single dataframe contains **129487 rows**

### Binary Logistic Regression

In [9]:
airline_data.drop(columns='satisfaction')

Gender      Customer Type  Age   Type of Travel     Class  \
0       Female     Loyal Customer   65  Personal Travel       Eco   
1         Male     Loyal Customer   47  Personal Travel  Business   
2       Female     Loyal Customer   15  Personal Travel       Eco   
3       Female     Loyal Customer   60  Personal Travel       Eco   
4       Female     Loyal Customer   70  Personal Travel       Eco   
...        ...                ...  ...              ...       ...   
129875  Female  disloyal Customer   29  Personal Travel       Eco   
129876    Male  disloyal Customer   63  Personal Travel  Business   
129877    Male  disloyal Customer   69  Personal Travel       Eco   
129878    Male  disloyal Customer   66  Personal Travel       Eco   
129879  Female  disloyal Customer   38  Personal Travel       Eco   

        Flight Distance  Seat comfort  Departure/Arrival time convenient  \
0                   265             0                                  0   
1                  2464             0                                  0   
2                  2138             0                                  0   
3                   623             0                                  0   
4                   354             0                                  0   
...                 ...           ...                                ...   
129875             1731             5                                  5   
129876             2087             2                                  3   
129877             2320             3                                  0   
129878             2450             3                                  2   
129879             4307             3                                  4   

        Food and drink  Gate location  ...  Online support  \
0                    0              2  ...               2   
1                    0              3  ...               2   
2                    0              3  ...               2   
3                    0              3  ...               3   
4                    0              3  ...               4   
...                ...            ...  ...             ...   
129875               5              3  ...               2   
129876               2              4  ...               1   
129877               3              3  ...               2   
129878               3              2  ...               2   
129879               3              3  ...               3   

        Ease of Online booking  On-board service  Leg room service  \
0                            3                 3                 0   
1                            3                 4                 4   
2                            2                 3                 3   
3                            1                 1                 0   
4                            2                 2                 0   
...                        ...               ...               ...   
129875                       2                 3                 3   
129876                       3                 2                 3   
129877                       4                 4                 3   
129878                       3                 3                 2   
129879                       4                 5                 5   

        Baggage handling  Checkin service  Cleanliness  Online boarding  \
0                      3                5            3                2   
1                      4                2            3                2   
2                      4                4            4                2   
3                      1                4            1                3   
4                      2                4            2                5   
...                  ...              ...          ...              ...   
129875                 4                4            4                2   
129876                 3                1            2                1   
12987

In [10]:
airline_data.dtypes

satisfaction                          object
Gender                                object
Customer Type                         object
Age                                    int64
Type of Travel                        object
Class                                 object
Flight Distance                        int64
Seat comfort                           int64
Departure/Arrival time convenient      int64
Food and drink                         int64
Gate location                          int64
Inflight wifi service                  int64
Inflight entertainment                 int64
Online support                         int64
Ease of Online booking                 int64
On-board service                       int64
Leg room service                       int64
Baggage handling                       int64
Checkin service                        int64
Cleanliness                            int64
Online boarding                        int64
Departure Delay in Minutes             int64
Arrival De

#### There are __22 independent variables/columns__ in the dataset.
#### We have 4 categorical features in the given dataset namely: __Gender, Customer Type, Type of Travel and Class__

In [11]:
print(set(airline_data['Gender']))
print(set(airline_data['Customer Type']))
print(set(airline_data['Type of Travel']))
print(set(airline_data['Class']))

{'Female', 'Male'}
{'Loyal Customer', 'disloyal Customer'}
{'Personal Travel', 'Business travel'}
{'Eco', 'Business', 'Eco Plus'}


In [12]:
#One hot encoding function
from sklearn.preprocessing import OneHotEncoder
def get_ohe(df, col):
    ohe = OneHotEncoder(drop='first', handle_unknown='error', sparse_output=False, dtype='int')
    ohe.fit(df[[col]])
    temp_df = pd.DataFrame(data=ohe.transform(df[[col]]), columns=ohe.get_feature_names_out())
    df.drop(columns=[col], axis=1, inplace=True)
    df = pd.concat([df.reset_index(drop=True), temp_df], axis=1)
    return df

In [13]:
airline_data = get_ohe(airline_data, 'Customer Type')
airline_data = get_ohe(airline_data, 'Class')
airline_data = get_ohe(airline_data, 'Type of Travel')
airline_data = get_ohe(airline_data, 'Gender')
airline_data.drop(columns='satisfaction')

Age  Flight Distance  Seat comfort  Departure/Arrival time convenient  \
0        65              265             0                                  0   
1        47             2464             0                                  0   
2        15             2138             0                                  0   
3        60              623             0                                  0   
4        70              354             0                                  0   
...     ...              ...           ...                                ...   
129482   29             1731             5                                  5   
129483   63             2087             2                                  3   
129484   69             2320             3                                  0   
129485   66             2450             3                                  2   
129486   38             4307             3                                  4   

        Food and drink  Gate location  Inflight wifi service  \
0                    0              2                      2   
1                    0              3                      0   
2                    0              3                      2   
3                    0              3                      3   
4                    0              3                      4   
...                ...            ...                    ...   
129482               5              3                      2   
129483               2              4                      2   
129484               3              3                      3   
129485               3              2                      3   
129486               3              3                      3   

        Inflight entertainment  Online support  Ease of Online booking  ...  \
0                            4               2                       3  ...   
1                            2               2                       3  ...   
2                            0               2                       2  ...   
3                            4               3                       1  ...   
4                            3               4                       2  ...   
...                        ...             ...                     ...  ...   
129482                       5               2                       2  ...   
129483                       1               1                       3  ...   
129484                       2               2                       4  ...   
129485                       2               2                       3  ...   
129486                       3               3                       4  ...   

        Checkin service  Cleanliness  Online boarding  \
0                     5            3                2   
1                     2            3                2   
2                     4            4                2   
3                     4            1                3   
4                     4            2                5   
...                 ...          ...              ...   
129482                4            4                2   
129483                1            2                1   
129484                2            3                2   
129485                2            1                2   
129486                3            3                3   

        Departure Delay in Minutes  Arrival Delay in Minutes  \
0                                0                      0.00   
1                              310                    305.00   
2                                0                      0.00   
3                                0                      0.00   
4                                0                      0.00   
...                            ...                       ...   
129482                           0                      0.00   
129483                         174                    172.00   
129484                         155                    16

#### We get **23 features(independent variables)** after performing One Hot Encoding

In [14]:
#Train and Test data splitting
X_train, X_test, y_train, y_test = train_test_split(airline_data.drop(columns = ['satisfaction']), airline_data['satisfaction'], test_size=0.2, stratify = airline_data['satisfaction'], random_state=50)
X_train
X_test
y_train
y_test

Age  Flight Distance  Seat comfort  Departure/Arrival time convenient  \
77869    44             2908             3                                  2   
59474    30             2105             4                                  4   
119053   59             3106             1                                  1   
84136    58             3234             3                                  3   
8209     60             1673             2                                  2   
...     ...              ...           ...                                ...   
75705    49             2363             2                                  1   
63094    38             1930             5                                  5   
98210    45             3005             1                                  1   
96020    45             3802             3                                  3   
119246   39             3784             4                                  4   

        Food and drink  Gate location  Inflight wifi service  \
77869                4              2                      5   
59474                4              3                      3   
119053               1              1                      5   
84136                3              3                      3   
8209                 3              2                      4   
...                ...            ...                    ...   
75705                1              1                      2   
63094                5              4                      1   
98210                1              1                      5   
96020                3              3                      5   
119246               4              4                      3   

        Inflight entertainment  Online support  Ease of Online booking  ...  \
77869                        4               4                       3  ...   
59474                        4               3                       3  ...   
119053                       5               4                       5  ...   
84136                        4               5                       3  ...   
8209                         4               3                       4  ...   
...                        ...             ...                     ...  ...   
75705                        2               2                       3  ...   
63094                        5               1                       1  ...   
98210                        4               5                       4  ...   
96020                        4               4                       4  ...   
119246                       5               4                       5  ...   

        Checkin service  Cleanliness  Online boarding  \
77869                 3            3                4   
59474                 4            5                3   
119053                4            5                3   
84136                 5            3                3   
8209                  5            4                4   
...                 ...          ...              ...   
75705                 2            1                2   
63094                 4            4                1   
98210                 5            4                4   
96020                 3            4                4   
119246                3            5                5   

        Departure Delay in Minutes  Arrival Delay in Minutes  \
77869                           60                     30.00   
59474                           24                     27.00   
119053                           0                      0.00   
84136                           57                     43.00   
8209                             8                      9.00   
...                            ...                       ...   
75705                          185                    196.00   
63094                           32                     30.00   
98210                           36                     4

Age  Flight Distance  Seat comfort  Departure/Arrival time convenient  \
126910   39             1861             1                                  1   
113265   19             1126             5                                  1   
95461    56              433             5                                  5   
82721    52             3949             2                                  2   
32709    67             2147             4                                  2   
...     ...              ...           ...                                ...   
125165   52             1403             5                                  5   
9482     68             2047             2                                  2   
72383    41             2236             2                                  2   
50139    11             1811             2                                  4   
48808    22             1842             2                                  2   

        Food and drink  Gate location  Inflight wifi service  \
126910               1              1                      3   
113265               1              1                      5   
95461                5              5                      2   
82721                2              2                      5   
32709                4              2                      1   
...                ...            ...                    ...   
125165               5              5                      5   
9482                 2              2                      3   
72383                2              2                      3   
50139                3              4                      1   
48808                2              4                      3   

        Inflight entertainment  Online support  Ease of Online booking  ...  \
126910                       5               4                       5  ...   
113265                       5               4                       5  ...   
95461                        4               4                       4  ...   
82721                        4               5                       3  ...   
32709                        4               1                       1  ...   
...                        ...             ...                     ...  ...   
125165                       5               5                       5  ...   
9482                         2               3                       3  ...   
72383                        3               4                       2  ...   
50139                        3               4                       1  ...   
48808                        2               1                       3  ...   

        Checkin service  Cleanliness  Online boarding  \
126910                4            5                5   
113265                4            4                5   
95461                 4            4                3   
82721                 4            3                4   
32709                 4            2                1   
...                 ...          ...              ...   
125165                3            5                5   
9482                  1            2                3   
72383                 1            2                4   
50139                 4            3                1   
48808                 3            3                3   

        Departure Delay in Minutes  Arrival Delay in Minutes  \
126910                           8                      2.00   
113265                           0                      5.00   
95461                            0                      0.00   
82721                           11                      9.00   
32709                            0                      0.00   
...                            ...                       ...   
125165                          66                     63.00   
9482                             5                      5.00   
72383                            0                      

77869     dissatisfied
59474        satisfied
119053       satisfied
84136        satisfied
8209         satisfied
              ...     
75705     dissatisfied
63094        satisfied
98210        satisfied
96020        satisfied
119246       satisfied
Name: satisfaction, Length: 103589, dtype: object

126910       satisfied
113265       satisfied
95461        satisfied
82721        satisfied
32709     dissatisfied
              ...     
125165       satisfied
9482      dissatisfied
72383     dissatisfied
50139     dissatisfied
48808     dissatisfied
Name: satisfaction, Length: 25898, dtype: object

#### length of training data = __103589__
#### length of testing data  = __25898__
#### first row's index for the train data = __77869__
#### first row's index for the test data = __126910__


In [15]:
#Scaling the data
if True: 
    from sklearn.preprocessing import StandardScaler
    sc = StandardScaler()
    X_train = pd.DataFrame(sc.fit_transform(X_train), columns = X_train.columns, index = X_train.index)
    X_test = pd.DataFrame(sc.transform(X_test), columns = X_test.columns, index = X_test.index)
    X_train
    X_test
    y_train
    y_test

Age  Flight Distance  Seat comfort  \
77869   0.30             0.90          0.12   
59474  -0.62             0.12          0.83   
119053  1.29             1.09         -1.32   
84136   1.23             1.22          0.12   
8209    1.36            -0.30         -0.60   
...      ...              ...           ...   
75705   0.63             0.37         -0.60   
63094  -0.10            -0.05          1.55   
98210   0.37             1.00         -1.32   
96020   0.37             1.77          0.12   
119246 -0.03             1.75          0.83   

        Departure/Arrival time convenient  Food and drink  Gate location  \
77869                               -0.65            0.80          -0.76   
59474                                0.66            0.80           0.01   
119053                              -1.30           -1.28          -1.53   
84136                                0.01            0.10           0.01   
8209                                -0.65            0.10          -0.76   
...                                   ...             ...            ...   
75705                               -1.30           -1.28          -1.53   
63094                                1.32            1.49           0.77   
98210                               -1.30           -1.28          -1.53   
96020                                0.01            0.10           0.01   
119246                               0.66            0.80           0.77   

        Inflight wifi service  Inflight entertainment  Online support  \
77869                    1.33                    0.46            0.37   
59474                   -0.19                    0.46           -0.40   
119053                   1.33                    1.20            0.37   
84136                   -0.19                    0.46            1.13   
8209                     0.57                    0.46           -0.40   
...                       ...                     ...             ...   
75705                   -0.95                   -1.03           -1.16   
63094                   -1.71                    1.20           -1.93   
98210                    1.33                    0.46            1.13   
96020                    1.33                    0.46            0.37   
119246                  -0.19                    1.20            0.37   

        Ease of Online booking  ...  Checkin service  Cleanliness  \
77869                    -0.36  ...            -0.27        -0.61   
59474                    -0.36  ...             0.52         1.12   
119053                    1.17  ...             0.52         1.12   
84136                    -0.36  ...             1.32        -0.61   
8209                      0.40  ...             1.32         0.25   
...                        ...  ...              ...          ...   
75705                    -0.36  ...            -1.06        -2.35   
63094                    -1.90  ...             0.52         0.25   
98210                     0.40  ...             1.32         0.25   
96020                     0.40  ...            -0.27         0.25   
119246                    1.17  ...            -0.27         1.12   

        Online boarding  Departure Delay in Minutes  Arrival Delay in Minutes  \
77869              0.50                        1.19                      0.38   
59474             -0.27                        0.24                      0.31   
119053            -0.27                       -0.39                     -0.39   
84136             -0.27                        1.11                      0.72   
8209               0.50                       -0.18                     -0.16   
...                 ...                         ...                       ...   
75705             -1.04                        4.47                      4.68   
63094             -1.81                        0.45                      0.38   
98210              0.50                        0.56                      0.85   
96020          

Age  Flight Distance  Seat comfort  \
126910 -0.03            -0.12         -1.32   
113265 -1.35            -0.83          1.55   
95461   1.09            -1.50          1.55   
82721   0.83             1.91         -0.60   
32709   1.82             0.16          0.83   
...      ...              ...           ...   
125165  0.83            -0.56          1.55   
9482    1.89             0.06         -0.60   
72383   0.10             0.25         -0.60   
50139  -1.88            -0.16         -0.60   
48808  -1.15            -0.13         -0.60   

        Departure/Arrival time convenient  Food and drink  Gate location  \
126910                              -1.30           -1.28          -1.53   
113265                              -1.30           -1.28          -1.53   
95461                                1.32            1.49           1.54   
82721                               -0.65           -0.59          -0.76   
32709                               -0.65            0.80          -0.76   
...                                   ...             ...            ...   
125165                               1.32            1.49           1.54   
9482                                -0.65           -0.59          -0.76   
72383                               -0.65           -0.59          -0.76   
50139                                0.66            0.10           0.77   
48808                               -0.65           -0.59           0.77   

        Inflight wifi service  Inflight entertainment  Online support  \
126910                  -0.19                    1.20            0.37   
113265                   1.33                    1.20            0.37   
95461                   -0.95                    0.46            0.37   
82721                    1.33                    0.46            1.13   
32709                   -1.71                    0.46           -1.93   
...                       ...                     ...             ...   
125165                   1.33                    1.20            1.13   
9482                    -0.19                   -1.03           -0.40   
72383                   -0.19                   -0.28            0.37   
50139                   -1.71                   -0.28            0.37   
48808                   -0.19                   -1.03           -1.93   

        Ease of Online booking  ...  Checkin service  Cleanliness  \
126910                    1.17  ...             0.52         1.12   
113265                    1.17  ...             0.52         0.25   
95461                     0.40  ...             0.52         0.25   
82721                    -0.36  ...             0.52        -0.61   
32709                    -1.90  ...             0.52        -1.48   
...                        ...  ...              ...          ...   
125165                    1.17  ...            -0.27         1.12   
9482                     -0.36  ...            -1.85        -1.48   
72383                    -1.13  ...            -1.85        -1.48   
50139                    -1.90  ...             0.52        -0.61   
48808                    -0.36  ...            -0.27        -0.61   

        Online boarding  Departure Delay in Minutes  Arrival Delay in Minutes  \
126910             1.27                       -0.18                     -0.34   
113265             1.27                       -0.39                     -0.26   
95461             -0.27                       -0.39                     -0.39   
82721              0.50                       -0.10                     -0.16   
32709             -1.81                       -0.39                     -0.39   
...                 ...                         ...                       ...   
125165             1.27                        1.35                      1.24   
9482              -0.27                       -0.25                     -0.26   
72383              0.50                       -0.39                     -0.39   
50139          

77869     dissatisfied
59474        satisfied
119053       satisfied
84136        satisfied
8209         satisfied
              ...     
75705     dissatisfied
63094        satisfied
98210        satisfied
96020        satisfied
119246       satisfied
Name: satisfaction, Length: 103589, dtype: object

126910       satisfied
113265       satisfied
95461        satisfied
82721        satisfied
32709     dissatisfied
              ...     
125165       satisfied
9482      dissatisfied
72383     dissatisfied
50139     dissatisfied
48808     dissatisfied
Name: satisfaction, Length: 25898, dtype: object

In [16]:
#Binary logistic regression model
model = LogisticRegression(fit_intercept = True, solver='lbfgs', multi_class = 'auto', penalty = None)

model.fit(X_train, y_train) 

# The following gives the mean accuracy on the given data and labels
accuracy_train = model.score(X_train, y_train)

# This is the coefficient Beta_1, ..., Beta_23
model.coef_

# This is the coefficient Beta_0
model.intercept_
print("The fraction of accurate prediction is: ",round(accuracy_train,4))

LogisticRegression(penalty=None)

array([[-0.10933188, -0.11289484,  0.40343082, -0.31087083, -0.31133854,
         0.14813228, -0.0951655 ,  0.91064184,  0.11763942,  0.304656  ,
         0.39189821,  0.28826001,  0.12069634,  0.37708687,  0.09094497,
         0.20885878,  0.13813465, -0.34225785, -0.76953739, -0.36740616,
        -0.20807181, -0.35546439, -0.48941764]])

array([0.35141685])

The fraction of accurate prediction is:  0.8351


#### The result is __0.8351, or 83.51%__, for the score, which represents the fraction of accurate predictions. This indicates a fairly good fit since the model accounts for __83%__ of the variability in the dependent variable.

#### Also, the __βˆ0, βˆ1, βˆ2 are 0.3514, -0.1093 and -0.1128__ respectively.

In [17]:
#Calculation of the accuracy of test dataset
test_output = pd.DataFrame(model.predict(X_test), index = X_test.index, columns = ['pred_Satisfaction'])
test_output = test_output.merge(y_test, left_index = True, right_index = True)
print(model.score(X_test, y_test))

0.8365896980461812


#### The accuracy of the model on test dataset is __83.66__

In [18]:
test_output.head()

pred_Satisfaction  satisfaction
126910         satisfied     satisfied
113265         satisfied     satisfied
95461          satisfied     satisfied
82721          satisfied     satisfied
32709       dissatisfied  dissatisfied

In [19]:
#Calculation of Probability for output variable satisfaction being 1
test_output['Probability'] =  model.predict_proba(X_test)[:, 1]
test_output.head()

pred_Satisfaction  satisfaction  Probability
126910         satisfied     satisfied         0.98
113265         satisfied     satisfied         0.98
95461          satisfied     satisfied         0.96
82721          satisfied     satisfied         0.70
32709       dissatisfied  dissatisfied         0.09

#### For the first index value of the test data frame, the chance that satisfaction is 1 is equal to __0.98 or 98%__.  